File: Output/Utility/Hooks/use-rich-text.hook.lua
-- Compiled with roblox-ts v2.2.0
local TS = _G[script]
local Rich = TS.import(script, TS.getModule(script, "@rbxts", "rich-text-stream").out).default
local useMemo = TS.import(script, TS.getModule(script, "@rbxts", "roact-hooked").src).useMemo
local function useRichText(factory, deps)
	return useMemo(function()
		return factory(Rich()):toString()
	end, deps)
end
return {
	useRichText = useRichText,
}

File: Source/Utility/Hooks/use-rich-text.hook.ts
import Rich from "@rbxts/rich-text-stream";

import { useMemo } from "@rbxts/roact-hooked";

type RichTextStream = ReturnType<typeof Rich>;

export function useRichText(factory: (Stream: RichTextStream) => RichTextStream, deps?: readonly unknown[] | undefined): string {
    return useMemo(() => factory(Rich()).toString(), deps);
};